shared/src/sources/balancer/graph_api.rs

```diff
@@ -43,10 +46,10 @@ impl BalancerSubgraphClient {
         )?))
     }
 
-    /// Retrieves the list of registered pools from the subgraph.
-    pub async fn get_weighted_pools(&self) -> Result<RegisteredWeightedPools> {
+    /// Retrieves the list of registered stable pools from the subgraph.
+    pub async fn get_registered_pools(&self) -> Result<RegisteredPools> {
```

Comment:
Comment says it returns registered stable pools but the type is RegisteredPools which internally stores
```rust
pub enum RegisteredPool {
    Weighted(RegisteredWeightedPool),
    Stable(RegisteredStablePool),
}
```
Again I am wondering if this could be represented in the types better if all of the results here must be Stable. Also should we call the function get_registered_stable_pools?

Reply:
That, I believe is just a typo in the comment. I explicitly removed those `get_specific` methods wherever possible in favour of `get_generic`